### PR TITLE
JCLOUDS-1457 - Add Clean Server operation to ServerApi.

### PR DESCRIPTION
**Pull request by trevorflanagan**
The intention of this PR is to add a useful operation for cleaning servers that have failed to be added. I will expand [NetworkDomainTearDown example](https://github.com/jclouds/jclouds-examples/blob/master/dimensiondata/src/main/java/org/jclouds/examples/dimensiondata/cloudcontrol/NetworkDomainTearDown.java) with a call to ```cleanServer```.

### CODE REVIEW

File: dimensiondata/src/main/java/org/jclouds/dimensiondata/cloudcontrol/domain/State.java [hunk 1]
```diff
@@ -22,7 +22,7 @@
 
 public enum State {
 
-   NORMAL, FAILED_ADD, FAILED_CHANGE, FAILED_DELETE, PENDING_DELETE, DELETED, UNRECOGNIZED, PENDING_CHANGE;
+   NORMAL, FAILED_ADD, FAILED_CHANGE, FAILED_DELETE, PENDING_DELETE, DELETED, UNRECOGNIZED, PENDING_CHANGE, PENDING_CLEAN, REQUIRES_SUPPORT;
```
Comment (trevorflanagan) [inline dimensiondata/src/main/java/org/jclouds/dimensiondata/cloudcontrol/domain/State.java]: new states required for clean server operation

File: dimensiondata/src/main/java/org/jclouds/dimensiondata/cloudcontrol/domain/Vlan.java [hunk 2]
```diff
@@ -35,6 +36,7 @@ public static Builder builder() {
 
    public abstract String name();
 
+   @Nullable
```
Comment (trevorflanagan) [inline dimensiondata/src/main/java/org/jclouds/dimensiondata/cloudcontrol/domain/Vlan.java]: This change is not directly related to this PR.

File: dimensiondata/src/main/java/org/jclouds/dimensiondata/cloudcontrol/features/ServerApi.java [hunk 3]
```diff
@@ -150,6 +150,17 @@ String cloneServer(@PayloadParam("id") String id, @PayloadParam("imageName") Str
...
+   @Named("server:cleanServer")
+   @POST
+   @Path("/cleanServer")
+   @Produces(MediaType.APPLICATION_JSON)
+   @MapBinder(BindToJsonPayload.class)
+   /**
+    * Operation for cleaning servers with FAILED_ADD state
+    * @see org.jclouds.dimensiondata.cloudcontrol.domain.State.FAILED_ADD
+    */
+   void cleanServer(@PayloadParam("id") String id);
```
Comment (trevorflanagan) [inline dimensiondata/src/main/java/org/jclouds/dimensiondata/cloudcontrol/features/ServerApi.java]: I have not added a live test for this particular operation as it would be difficult to setup in a live scenario.

File: dimensiondata/src/main/java/org/jclouds/dimensiondata/cloudcontrol/features/ServerApi.java [hunk 4]
```diff
@@ -150,6 +150,17 @@ String cloneServer(@PayloadParam("id") String id, @PayloadParam("imageName") Str
...
 
+   @Named("server:cleanServer")
+   @POST
+   @Path("/cleanServer")
+   @Produces(MediaType.APPLICATION_JSON)
+   @MapBinder(BindToJsonPayload.class)
+   /**
+    * Operation for cleaning servers with FAILED_ADD state
+    * @see org.jclouds.dimensiondata.cloudcontrol.domain.State.FAILED_ADD
+    */
```
Comment (nacx) [inline dimensiondata/src/main/java/org/jclouds/dimensiondata/cloudcontrol/features/ServerApi.java]: Move javadoc before the annotations.

**Comment by nacx** [comment]
Thanks! Just one minor. Once addressed I'll merge the PR.

**Comment by trevorflanagan** [comment]
Thanks for looking @nacx , change has been made and commits squashed.

**Comment by nacx** [comment]
Merged to [master](http://git-wip-us.apache.org/repos/asf/jclouds-labs/commit/54249bd3) and [2.1.x](http://git-wip-us.apache.org/repos/asf/jclouds-labs/commit/6b4aa2c8). Thanks!
